src/rotest/management/base_resource.py

```diff
@@ -52,6 +52,8 @@ class BaseResource(object):
 
     Attributes:
         DATA_CLASS (class): class of the resource's global data container.
+        INITIALIZE_IN_THREAD (bool): whether or not to validate and initialize
```

Comment:
Why on the sub-resource level? It doesn't make sense
It makes more sense at the parent level. The sub-resource doesn't know how efficient or thread-safe it's to run it in a thread